rust-runtime/smithy-types/src/instant/format.rs

```diff
@@ -436,23 +436,18 @@ pub mod iso_8601 {
             )
         };
 
+        // This is stated in the assumptions for RFC-3339. ISO-8601 allows for years
+        // between -99,999 and 99,999 inclusive, but RFC-3339 is bound between 0 and 9,999.
         assert!(
-            year.abs() <= 99_999,
-            "years greater than 5 digits are not supported by ISO-8601"
+            (0..=9_999).contains(&year),
```

Comment:
can you file an issue about this panic? this should really be a serialization failure, not a runtime panic but I think it might be a big refactoring to make this method fallible right now. The general principle is that the SDK should never panic.

Reply:
Filed: https://github.com/awslabs/smithy-rs/issues/494